app/ts/components/TransactionDetailsPage.tsx

```diff
@@ -0,0 +1,221 @@
+import { ComponentChildren } from 'preact'
+import { ethers } from 'ethers'
+import * as Layout from './Layout.js'
+import { useParams } from './HashRouter.js'
+import { assertUnreachable, calculateGasFee, isTransactionHash } from '../library/utilities.js'
+import { assertTransferStatus, transferStore } from '../store/transfer.js'
+import { useAsyncState } from '../library/preact-utilities.js'
+import * as Icon from "./Icon/index.js"
+
+export const TransactionDetailsPage = () => {
+	return (
+		<Layout.Page>
+			<Layout.Header />
+			<Layout.Body>
+				<PageTitle />
+				<Main />
+			</Layout.Body>
+			<Layout.Footer />
+		</Layout.Page>
+	)
+}
+
+const Main = () => {
+	const transfer = transferStore
+	const hash = extractTransactionHashFromParams()
+
+	if (!hash) return <TransactionDetailsInvalid />
```

Comment:
```suggestion
	if (hash === undefined) return <TransactionDetailsInvalid />
```
its dangerous to check value without comparison as many things are false in JS

Reply:
just pushed commit using `null` instead because `extractTransactionHashFromParams` is actually `0x${string} | null`